tests/wp-unit/tests/include/Core/Posts/Metabox/DeletePostsByCategoryMetaboxTest.php

```diff
@@ -0,0 +1,634 @@
+		$this->assertEquals( 80, count( $posts ) );
+
+		// call our method. Second batch limit_to 30 posts
+		$delete_options = array(
+			'post_type'      => array( 'post' ),
+			'selected_cats'  => array( $cat1 ),
+			'restrict'       => false,
+			'date_op'        => false,
+			'days'           => false,
+			'private'        => false,
+			'limit_to'       => 30,
+			'force_delete'   => false,
+		);
+		$this->metabox->delete( $delete_options );
+
+		// Assert that cat1 has 50 posts.
+		$posts = $this->get_posts_by_category( $cat1 );
+		$this->assertEquals( 50, count( $posts ) );
+	}
+
+	/**
+	 * Add tests to test deleting posts from All categories by custom post type.
+	 */
+	public function test_for_deleting_posts_from_all_categories_custom_post_type() {
+		$this->register_post_type_for_bulk_delete();
+		$taxonomy  = 'category';
+		$post_type = 'movie';
+
+		// Create two terms.
+		$term1 = $this->factory->term->create( array( 'name' => 'term1', 'taxonomy' => $taxonomy ) );
+		$term2 = $this->factory->term->create( array( 'name' => 'term2', 'taxonomy' => $taxonomy ) );
+
+		// Create two posts and assign the term.
+		$post1 = $this->factory->post->create( array( 'post_title' => 'post1', 'post_type' => $post_type ) );
+		$post2 = $this->factory->post->create( array( 'post_title' => 'post2', 'post_type' => $post_type ) );
+		wp_set_object_terms( $post1, $term1, $taxonomy );
+		wp_set_object_terms( $post2, $term2, $taxonomy );
+
+		// Assert that each terms has one post.
+		$posts_in_term1 = $this->get_posts_by_custom_term( $term1, $taxonomy, $post_type );
+		$posts_in_term2 = $this->get_posts_by_custom_term( $term2, $taxonomy, $post_type );
+
+		$this->assertEquals( 1, count( $posts_in_term1 ) );
+		$this->assertEquals( 1, count( $posts_in_term2 ) );
+
+		// call our method.
+		$delete_options = array(
+			'post_type'      => array( $post_type ),
+			'selected_cats'  => array( 'all' ),
+			'restrict'       => false,
+			'private'        => false,
+			'limit_to'       => false,
+			'force_delete'   => false,
+			'date_op'        => false,
+			'days'           => false,
+		);
+		$this->metabox->delete( $delete_options );
+
+		// Assert that each post status moved to trash.
+		$post1_status = get_post_status( $post1 );
+		$post2_status = get_post_status( $post2 );
+
+		$this->assertEquals( 'trash', $post1_status );
+		$this->assertEquals( 'trash', $post2_status );
+	
+		// Assert that each terms has no post.
+		$posts_in_term1 = $this->get_posts_by_custom_term( $term1, $taxonomy, $post_type );
+		$posts_in_term2 = $this->get_posts_by_custom_term( $term2, $taxonomy, $post_type );
+
+		$this->assertEquals( 0, count( $posts_in_term1 ) );
+		$this->assertEquals( 0, count( $posts_in_term2 ) );
+	}
+
+	/**
+	 * Add tests to test deleting posts from one categories by custom post type.
+	 */
+	public function test_for_deleting_posts_from_one_category_custom_post_type() {
+		$this->register_post_type_for_bulk_delete();
```

Comment:
Refactor this method into `register_post_type` and then pass the name of the post type that needs to be registered.

You can ignore the labels and just register only the post type that is passed to that method.